### PR TITLE
PROPN_NNPS lemmas -> no s

### PR DESCRIPTION
**Pull request by AngledLuffa**
Addresses, at least partially, issue https://github.com/UniversalDependencies/UD_English-EWT/issues/147

### CODE REVIEW

**Comment by AngledLuffa** [comment]
Also fixes a few demonyms which weren't in the original list I searched for

File: en_ewt-ud-dev.conllu [hunk 1]
```diff
@@ -21877,14 +21877,14 @@
 2	are	be	AUX	VBP	Mood=Ind|Tense=Pres|VerbForm=Fin	4	cop	4:cop	_
 3	open	open	ADJ	JJ	Degree=Pos	4	amod	4:amod	_
 4	minded	minded	ADJ	JJ	Degree=Pos	0	root	0:root	_
-5	thr	thr	ADV	RB	_	4	advmod	4:advmod	_
+5	thr	thr	ADV	RB	Typo=Yes	4	advmod	4:advmod	CorrectForm=there
```
Comment (nschneid) [inline en_ewt-ud-dev.conllu:21880]: lemma = there

File: not-to-release/sources/answers/20111106022931AAwpqXm_ans.xml.conllu [hunk 2]
```diff
@@ -2,7 +2,7 @@
 # sent_id = answers-20111106022931AAwpqXm_ans-0001
 # newpar id = answers-20111106022931AAwpqXm_ans-p0001
 # text = NZers: Have you decided who you're going to vote for?
-1	NZers	NZers	PROPN	NNPS	Number=Plur	5	vocative	5:vocative	SpaceAfter=No
+1	NZers	NZer	PROPN	NNPS	Number=Plur	5	vocative	5:vocative	SpaceAfter=No
```
Comment (nschneid) [inline not-to-release/sources/answers/20111106022931AAwpqXm_ans.xml.conllu:5]: Abbr=Yes?

**Comment by AngledLuffa** [comment]
Updated those changes, although obviously there's also a lot more
discussion going on.

**Comment by nschneid** [comment]
I think there's a token of "democrat" and a token of "republican" that should be capitalized in the lemma.

**Comment by AngledLuffa** [comment]
Done.

**Comment by nschneid** [comment]
Excellent! Now how about updating to United/NNP States/NNPS? This is actually specified in the PTB tagging guidelines:

image

**Comment by nschneid** [comment]
Also Number=Plur

**Comment by AngledLuffa** [comment]
Done

File: en_ewt-ud-train.conllu [hunk 3]
```diff
@@ -204448,7 +204448,7 @@
 5	to	to	ADP	IN	_	9	case	9:case	_
 6	the	the	DET	DT	Definite=Def|PronType=Art	9	det	9:det	_
 7	Youngstown	Youngstown	PROPN	NNP	Number=Sing	9	compound	9:compound	_
-8	Sports	Sports	PROPN	NNP	Number=Sing	9	compound	9:compound	_
+8	Sports	Sport	PROPN	NNP	Number=Sing	9	compound	9:compound	_
```
Comment (nschneid) [inline en_ewt-ud-train.conllu:204451]: NNPS, Number=Plur?

**Comment by AngledLuffa** [comment]
Yep

File: not-to-release/sources/answers/20111107193044AAvUYBv_ans.xml.conllu [hunk 4]
```diff
@@ -85,7 +85,7 @@
 29	mail	mail	X	NN	Number=Sing	28	goeswith	28:goeswith	_
 30	from	from	ADP	IN	_	32	case	32:case	_
 31	southwest	southwest	PROPN	NNP	Number=Sing	32	compound	32:compound	_
-32	airlines	airlines	PROPN	NNPS	Number=Plur	28	nmod	28:nmod:from	_
+32	airlines	airline	PROPN	NNPS	Number=Plur	28	nmod	28:nmod:from	_
```
Comment (nschneid) [inline not-to-release/sources/answers/20111107193044AAvUYBv_ans.xml.conllu:88]: capitalize lemmas for both PROPNs?

File: not-to-release/sources/answers/20111107203006AA9ojw8_ans.xml.conllu [hunk 5]
```diff
@@ -34,7 +34,7 @@
 16	back	back	ADV	RB	_	15	advmod	15:advmod	_
 17	to	to	ADP	IN	_	19	case	19:case	_
 18	the	the	DET	DT	Definite=Def|PronType=Art	19	det	19:det	_
-19	states	states	PROPN	NNP	Number=Sing	16	obl	16:obl:to	SpaceAfter=No
+19	states	State	PROPN	NNPS	Number=Sing	16	obl	16:obl:to	SpaceAfter=No
```
Comment (nschneid) [inline not-to-release/sources/answers/20111107203006AA9ojw8_ans.xml.conllu:37]: Number=Plur

File: not-to-release/sources/answers/20111107221352AAlIioO_ans.xml.conllu [hunk 6]
```diff
@@ -130,7 +130,7 @@
 # sent_id = answers-20111107221352AAlIioO_ans-0009
 # text = prime ribs have very good food but its super expensive
 1	prime	prime	PROPN	NNP	Number=Sing	2	compound	2:compound	_
-2	ribs	ribs	PROPN	NNPS	Number=Plur	3	nsubj	3:nsubj	_
+2	ribs	rib	PROPN	NNPS	Number=Plur	3	nsubj	3:nsubj	_
```
Comment (nschneid) [inline not-to-release/sources/answers/20111107221352AAlIioO_ans.xml.conllu:133]: capitalize lemmas?

File: not-to-release/sources/answers/20111108085945AAgJhOG_ans.xml.conllu [hunk 7]
```diff
@@ -236,7 +236,7 @@
 4	s	's	PART	POS	_	3	case	3:case	_
 5	aquarium	aquarium	PROPN	NNP	Number=Sing	0	root	0:root	_
 6	and	and	CCONJ	CC	_	7	cc	7:cc	_
-7	pets	pets	PROPN	NNPS	Number=Plur	5	conj	5:conj:and	_
+7	pets	pet	PROPN	NNPS	Number=Plur	5	conj	5:conj:and	_
```
Comment (nschneid) [inline not-to-release/sources/answers/20111108085945AAgJhOG_ans.xml.conllu:239]: capitalize PROPN lemmas?

File: not-to-release/sources/answers/20111108093942AAYF9Dn_ans.xml.conllu [hunk 8]
```diff
@@ -21,7 +21,7 @@
 # newpar id = answers-20111108093942AAYF9Dn_ans-p0002
 # text = UP Roadways buses ply frequently between Sector-37 crossing in Noida and Kasna in Greater Noida, passing through Pari Chowk.
 1	UP	UP	PROPN	NNP	Number=Sing	2	compound	2:compound	_
-2	Roadways	Roadways	PROPN	NNP	Number=Sing	3	compound	3:compound	_
+2	Roadways	Roadway	PROPN	NNP	Number=Sing	3	compound	3:compound	_
```
Comment (nschneid) [inline not-to-release/sources/answers/20111108093942AAYF9Dn_ans.xml.conllu:24]: Number=Plur

Comment (nschneid) [reply not-to-release/sources/answers/20111108093942AAYF9Dn_ans.xml.conllu:24]: and NNPS

File: not-to-release/sources/answers/20111108102205AArwNzY_ans.xml.conllu [hunk 9]
```diff
@@ -176,7 +176,7 @@
 6	here	here	ADV	RB	PronType=Dem	5	advmod	5:advmod	_
 7	from	from	ADP	IN	_	9	case	9:case	_
 8	the	the	DET	DT	Definite=Def|PronType=Art	9	det	9:det	_
-9	states	states	PROPN	NNP	Number=Sing	5	nmod	5:nmod:from	_
+9	states	State	PROPN	NNPS	Number=Sing	5	nmod	5:nmod:from	_
```
Comment (nschneid) [inline not-to-release/sources/answers/20111108102205AArwNzY_ans.xml.conllu:179]: Number=Plur

**Review by nschneid — CHANGES_REQUESTED** [review]
Looked over a bunch. A lot of great improvements. Just a couple things:

1. Please use a script to ensure that NNPS == Number=Plur == form ends in -s but lemma doesn't. And NNP == Number=Sing. And PROPN == (NNP or NNPS). I think those are all correct assumptions for now.
2. I don't understand the policy for capitalizing PROPN lemmas of places/organizations where the form is lowercase. There seem to be inconsistencies, even for pluralized names at issue here. (I realize that standardizing this for non-pluralized names would be out of scope.)

**Comment by amir-zeldes** [comment]
Just as a note, GUM does capitalize things like southwestern airlines -> Southwestern Airline:

https://corpling.uis.georgetown.edu/annis/#_q=eHBvcz0vTk5QLj8vIF89XyB0b2s9L1thLXpdKy8gXz1fIGxlbW1hPS9bQS1aXS4qLw&_c=R1VN&cl=5&cr=5&s=0&l=10

**Comment by AngledLuffa** [comment]
I fixed each of the specific things you caught.

I did the following greps:

> NNPS == Number=Plur ...
> And NNP == Number=Sing

grep NNP * | grep "Number=Plur" | grep -v NNPS
grep NNPS * | grep "Number=Sing"

> PROPN == (NNP or NNPS)

grep PROPN * | grep -v NNP

This one turned up Central, which is tagged slightly inconsistently.  I see
no reason to consider Central Iowa a proper noun - I don't think this is a
county name or that the regions have specific names, although I could be
wrong.  Central Otago is in fact the name of a region in NZ.  What about
Central Intelligence Agency?  In cases such as North Vietnam, we label
North as PROPN/NNP, so I guess Central should also be PROPN/NNP?  That's
not a compelling argument to me, though.  There's also Central Asian
Republics, for which various words can be found capitalized if you search
online.

It also dredged up No. which I suppose should be labeled Abbr=Yes
everywhere.

Modern China?  modern as the lemma with ADJ/JJ as the tag?  This specific
example is part of a book title

> NNPS == Number=Plur == form ends in -s

Disagree...  plural nouns don't necessarily end in s.  People, for
example.  Although the exceptions are pretty few and can be checked by hand.

egrep "[a-rt-zA-RT-Z]  [a-zA-Z]+       PROPN   NNPS" *
(those are tab characters, not spaces)

Is the lemma of People person?  What if it's a collective noun, such as The
People vs EWT?

What is the tag on Chinese instead of Chinese people?  It sounds like a
noun to me, but perhaps it's an adjective with a hidden noun.

> ... but lemma doesn't

grep "s        PROPN   NNPS" *
As we've been over, there's a whole lot of exceptions.  Falls, Beatles,
Valdes, ... Mathematics?  I suggest punting on this and making a new issue
if needed

**Comment by nschneid** [comment]
Central Intelligence Agency: Central/PROPN is fine for now. (We'll revisit tagging of adjectives in names separately. I was mainly concerned about lowercase forms in plural names being given uppercase lemmas only some of the time.)

As in WordNet, it should be people/people/Sing if referring to a specific body (a people) and people/person/Plur otherwise. I would say the former applies to the political uses and the latter to "Highly Effective People".

If Chinese heads a noun phrase, e.g. "I had Chinese last night" (implicit: food), I would guess it should be PROPN.

**Comment by AngledLuffa** [comment]
Now the Electoral Commission is refusing to punish people for mere past
Baath Party membership.

--- person?

 They are setting the road blocks at the exact same positions they were
during the war last year, which indicates they are the same people

--- person as well?

 people who are suggesting that we go out and fight them are living in
dream land.

--- also sounds like person

Pakistan People's Party

--- People?  people?

 the people of Pakistan

--- people?  although it's tagged NNS in the treebank

Actually this is becoming really tedious.  Suggest opening a separate issue
for this as well instead of fixing it in this changelist

**Comment by nschneid** [comment]
> Actually this is becoming really tedious. 

Yeah I would limit this PR to proper names.

File: en_ewt-ud-train.conllu [hunk 10]
```diff
@@ -218496,41 +218496,41 @@
...
 21	and	and	CCONJ	CC	_	23	cc	23:cc	_
 22	for	for	ADP	IN	_	23	case	23:case	_
 23	game	game	NOUN	NN	Number=Sing	11	conj	5:obl:for|11:conj:and	_
 24	4	4	NUM	CD	NumType=Card	23	nummod	23:nummod	_
 25	of	of	ADP	IN	_	28	case	28:case	_
 26	the	the	DET	DT	Definite=Def|PronType=Art	28	det	28:det	_
-27	world	world	NOUN	NN	Number=Sing	28	compound	28:compound	_
-28	series	series	NOUN	NNS	Number=Plur	23	nmod	23:nmod:of	SpaceAfter=No
+27	world	World	PROPN	NNP	Number=Sing	28	compound	28:compound	_
+28	series	Series	PROPN	NNP	Number=Plur	23	nmod	23:nmod:of	SpaceAfter=No
```
Comment (nschneid) [inline en_ewt-ud-train.conllu:218507]: Number=Sing

File: en_ewt-ud-train.conllu [hunk 11]
```diff
@@ -26368,7 +26368,7 @@
 3	Diary	Diary	PROPN	NNP	Number=Sing	0	root	0:root	_
 4	about	about	ADP	IN	_	7	case	7:case	_
 5-6	bush's	_	_	_	_	_	_	_	_
-5	bush	bush	PROPN	NNPS	Number=Plur	7	nmod:poss	7:nmod:poss	_
+5	bush	Bush	PROPN	NNP	Number=Plur	7	nmod:poss	7:nmod:poss	_
```
Comment (nschneid) [inline en_ewt-ud-train.conllu:26371]: Number=Sing

File: en_ewt-ud-train.conllu [hunk 12]
```diff
@@ -131625,7 +131625,7 @@
 16	to	to	ADP	IN	_	22	case	22:case	_
 17	the	the	DET	DT	Definite=Def|PronType=Art	18	det	18:det	_
 18-19	Devil's	_	_	_	_	_	_	_	_
-18	Devil	Devil	PROPN	NNPS	Number=Plur	22	nmod:poss	22:nmod:poss	_
+18	Devil	Devil	PROPN	NNP	Number=Plur	22	nmod:poss	22:nmod:poss	_
```
Comment (nschneid) [inline en_ewt-ud-train.conllu:131628]: Number=Sing

File: en_ewt-ud-train.conllu [hunk 13]
```diff
@@ -146475,29 +146475,29 @@
 42	which	which	PRON	WDT	PronType=Rel	46	nsubj	41:ref	_
 43	was	be	AUX	VBD	Mood=Ind|Number=Sing|Person=3|Tense=Past|VerbForm=Fin	46	cop	46:cop	_
 44-45	mcallister's	_	_	_	_	_	_	_	_
-44	mcallister	mcallister	PROPN	NNPS	Number=Plur	46	nmod:poss	46:nmod:poss	_
+44	mcallister	McAllister	PROPN	NNP	Number=Plur	46	nmod:poss	46:nmod:poss	_
```
Comment (nschneid) [inline en_ewt-ud-train.conllu:146478]: Number=Sing

File: en_ewt-ud-train.conllu [hunk 14]
```diff
@@ -146475,29 +146475,29 @@
...
 45	's	's	PART	POS	_	44	case	44:case	_
-46	deli	deli	PROPN	NNPS	Number=Plur	41	acl:relcl	41:acl:relcl	_
+46	deli	Deli	PROPN	NNPS	Number=Plur	41	acl:relcl	41:acl:relcl	_
```
Comment (nschneid) [inline en_ewt-ud-train.conllu:146480]: NNP, Number=Sing?

**Comment by amir-zeldes** [comment]
> As in WordNet, it should be people/people/Sing if referring to a specific body (a people) and people/person/Plur otherwise. I would say the former applies to the political uses and the latter to "Highly Effective People".

FYI in GUM it has been people -> people regardless of sense and number, treating person/persons as a separate lemma. Aside from WordNet is there a lemmatized corpus precedent advocating people -> person?

**Comment by nschneid** [comment]
I don't know about lemmatization practices in corpora that don't follow WordNet but it's certainly the case that when we want to pluralize "person" we usually say "people"—"persons" is restricted to certain genres.

**Comment by amir-zeldes** [comment]
OK, I can match that in GUM too

**Pull request merged**